### PR TITLE
[receiver/googlecloudpubsub] allow endpoint and insecure config

### LINKED ISSUE #10844: googlecloudpubsub receiver dropped support for pubsub Emulator
**Describe the bug**
Pubsub emulator is not supported anymore by the receiver, as the `endpoint`  and `insecure` configurations were made private here https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8391/files#diff-c3c2b69e4538d56891424468d9ffd5bd0dfb2cf4f0d7bbcfddd23b9f901d6d09

Pubsub Emulator support is crucial for automation tests and local development pipelines.

**Steps to reproduce**
Start the pubsub emulator by running `gcloud beta emulators pubsub start` (more info on the [official guide](https://cloud.google.com/pubsub/docs/emulator)) and configure the googlecloudpubsub receiver using `endpoint: localhost:8085`

**What did you expect to see?**
Collector connecting correctly to the pubsub subscription of the emulator.

**What did you see instead?**

```Error: '' has invalid keys: endpoint, insecure```

**What version did you use?**
Version: 0.53.0

**What config did you use?**
```
receivers:
  googlecloudpubsub:
    project: "my-project-id"
    endpoint: localhost:8085
    insecure: true
```

**Additional context**

### PR DESCRIPTION
**Pull request by k7k0**
**Description:** <Describe what has changed.>
Fixes #10844 allowing to connect the receiver to a local PubSub emulator instance or switching between [global and regional service endpoints](https://cloud.google.com/pubsub/docs/reference/service_apis_overview#service_endpoints)

**Link to tracking Issue:** #10844

**Testing:** tested locally by [running  the emulator](https://cloud.google.com/pubsub/docs/emulator) and configuring the receiver with
```
receivers:
  googlecloudpubsub:
    project: "project-id"
    endpoint: localhost:8085
    insecure: true
```

**Documentation:** Added config doc for both params

### CODE REVIEW

File: receiver/googlecloudpubsubreceiver/README.md
Comment (alexvanboxel) [inline receiver/googlecloudpubsubreceiver/README.md:24]: never used the regional endpoints, so I tried the out of curiosity. Verified the branch locally and worked (as well with a regional endpoint)

**Review by alexvanboxel — APPROVED** [review]

**Comment by alexvanboxel** [comment]
@djaglowski can you kick off the tests? thanks.

**Review by alexvanboxel — CHANGES_REQUESTED** [review]

**Comment by k7k0** [comment]
fixed tests. @djaglowski can you kick off again? thanks.

**Review by alexvanboxel — APPROVED** [review]

**Comment by djaglowski** [comment]
This looks good to me, but would appreciate a final review from @alexvanboxel before I merge it.

**Comment by alexvanboxel** [comment]
> This looks good to me, but would appreciate a final review from @alexvanboxel before I merge it.

already did and tested. Tests give a changelog error though

**Comment by djaglowski** [comment]
@k7k0, please add a changelog entry.

**Review by alexvanboxel — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Comment by alexvanboxel** [comment]
all green, now only need approval from allowed approvers

**Pull request merged**
